sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -125,6 +133,7 @@ public PCollectionConsumerRegistry(
 
     pCollectionIdsToConsumers.put(
         pCollectionId, ConsumerAndMetadata.forConsumer(consumer, pTransformId, state));
+    pCollectionIdsToCoders.put(pCollectionId, coder);
```

Comment:
How about having `ConsumerAndMetadata ` to include coder information instead of having a separated mapping?